### PR TITLE
Get normalized headers in all the notebooks

### PR DESCRIPTION
**Pull request by dangunter**
We have been talking for a while about having standard header information in our example notebooks. This PR is an attempt to 

- [x] enforce standard headers (`idaes_examples.tests.test_notebooks.test_header`
- [x] provide some tools (`idaesx hdr`) to help fix and maintain metadata
- [x]  fix the headers in all the existing notebooks

----
Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

    I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
    I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

----
:books: Documentation preview :books:: https://idaes-examples--57.org.readthedocs.build/en/57/

### CODE REVIEW

**Comment by dangunter** [comment]
The initial pass is to expect an author and maintainer in all notebooks. The format is dead simple. In the cell with the notebook title, typically the first one after the copyright, put author/maintainer like this
```
# Title of notebook
Author: Person Name
Maintainer: Another Person Name
```

Currently most notebooks do not have this of course. The test looks for this and fails if any notebook is missing author/maintainer. It gives a list of all offending notebooks, currently:
```
[[ 31 errors ]] base: C:\Users\dkgun\src\dangunter\idaes-examples\idaes_examples\notebooks

 1) docs\diagnostics\degeneracy_hunter.ipynb: Missing required metadata keys: author,maintainer
 2) docs\dae\petsc_chem.ipynb: Missing required metadata keys: author,maintainer
 3) docs\dae\petsc_pid.ipynb: Missing required metadata keys: author,maintainer
 4) docs\flowsheets\hda_flowsheet_with_costing.ipynb: Missing required metadata keys: author,maintainer
 5) docs\flowsheets\hda_flowsheet_with_distillation.ipynb: Missing required metadata keys: author,maintainer
 6) docs\flowsheets\methanol_synthesis.ipynb: Missing required metadata keys: author,maintainer
 7) docs\flowsheets\CO2_adsorption_desorption\CO2_Adsorption_Desorption_1DFixedBed.ipynb: Missing required metadata keys: author,maintainer
 8) docs\unit_models\operations\compressor.ipynb: Missing required metadata keys: author,maintainer
 9) docs\unit_models\operations\heater.ipynb: Missing required metadata keys: author,maintainer
10) docs\unit_models\operations\heat_exchanger_0d.ipynb: Missing required metadata keys: author,maintainer
11) docs\unit_models\operations\mixer.ipynb: Missing required metadata keys: author,maintainer
12) docs\unit_models\operations\pump.ipynb: Missing required metadata keys: author,maintainer
13) docs\unit_models\operations\skeleton_unit.ipynb: Missing required metadata keys: author,maintainer
14) docs\unit_models\operations\turbine.ipynb: Missing required metadata keys: author,maintainer
15) docs\unit_models\reactors\cstr.ipynb: Missing required metadata keys: author,maintainer
16) docs\unit_models\reactors\equilibrium_reactor.ipynb: Missing required metadata keys: author,maintainer
17) docs\unit_models\reactors\gibbs_reactor.ipynb: Missing required metadata keys: author,maintainer
18) docs\unit_models\reactors\plug_flow_reactor.ipynb: Missing required metadata keys: author,maintainer
19) docs\unit_models\reactors\stoichiometric_reactor.ipynb: Missing required metadata keys: author,maintainer
20) docs\unit_models\custom_unit_models\custom_compressor.ipynb: Missing required metadata keys: author,maintainer
21) docs\unit_models\custom_unit_models\custom_heater.ipynb: Missing required metadata keys: author,maintainer
22) docs\properties\dictionary_txy_diagrams.ipynb: Missing required metadata keys: author,maintainer
23) docs\properties\parameter_estimation_pr.ipynb: Missing required metadata keys: author,maintainer
24) docs\properties\custom\custom_physical_property_packages.ipynb: Missing required metadata keys: author,maintainer
25) docs\properties\custom\custom_reaction_property_packages.ipynb: Missing required metadata keys: author,maintainer
26) docs\surrogates\best_practices_optimization.ipynb: Missing required metadata keys: author,maintainer
27) docs\surrogates\pysmo\pysmo_basics.ipynb: Missing required metadata keys: author,maintainer
28) docs\surrogates\pysmo\pysmo_flowsheet_optimization.ipynb: Missing required metadata keys: author,maintainer
29) docs\surrogates\alamo\alamo_flowsheet_optimization.ipynb: Missing required metadata keys: author,maintainer
30) docs\surrogates\omlt\keras_flowsheet_optimization.ipynb: Missing required metadata keys: author,maintainer
31) active\power_gen\ngcc\ngcc.ipynb: Missing required metadata keys: author,maintainer
```

**Comment by dangunter** [comment]
@andrewlee94 @lbianchi-lbl comments welcome on goals and implementation here

**Comment by andrewlee94** [comment]
@dangunter I think this is a good idea - the biggest challenge will be keeping this up-to-date. I doubt this is possible, but it would be nice if we could somehow check that the header values are correct on a PR; e.g. check that the person submitting the PR is listed as a maintainer.

Something else that might be useful (but again hard to do) would be to have a "last updated" field so that user (and project admins) can see how recently an example has been updated (and by corollary, how likely it is to be unsupported).

**Comment by dangunter** [comment]
Now that I have programmatic updates in the code (see that 'idaesx hdr' command) I think we could do the last-updated at least. I don't know about changing author/maintainer -- just fixing a notebook or adding to it may not qualify.

So any ideas on how to assign that whole list. Easiest thing would be to loop through and assign them all to the same person but I don't think anyone really wants that.

**Comment by dangunter** [comment]
So a separate question: Should I add/require the "Updated: <date>" metadata? Wouldn't be hard to add the field, though I think I'd need to do some archaeology to figure out how to populate it

**Comment by andrewlee94** [comment]
@dangunter I'd go with the v2.1 release date; we might not have changed anything but we did check that they all ran as we did the transition so we know they are current as of then.

**Comment by andrewlee94** [comment]
As for owners, I think the place to start is to assign a first point of contact by category and get them to suggest maintainers for each notebook. As a start, I think the following are the people to ask for each folder:

1. docs\diagnostics\degeneracy_hunter.ipynb: Andrew Lee
2. docs\dae\ John Eslick
3. docs\flowsheets\ Miguel/Brandon
4. docs\unit_models\ Brandon
5. docs\unit_models\custom_unit_models\ Me (for lack of anyone better)
6. docs\properties\ Me and Alejandro
7. docs\surrogates\ Brandon/Miguel
8. docs\surrogates\pysmo\ Mayo
9. docs\surrogates\alamo\ Brandon/Miguel
10. docs\surrogates\omlt\ Carl/ (and Brandon as Carl will not respond)
11. active\power_gen\ngcc\ John Eslick

**Comment by dangunter** [comment]
@andrewlee94 thanks, got it. Will do that next. Have added "Updated:" to all the notebooks and also tried to make the 'header' cells all somewhat descriptive (often by merging the 2nd cell up with the 1st) so the "idaesx gui" browser can show it a decent preview. We probably should add an example and tutorial 'template' notebook for people to copy and extend -- can also add this to the "idaesx new" command but I expect this to get spotty usage, most people will just copy the notebook file and go.

**Comment by dangunter** [comment]
Created a spreadsheet to fill in author and maintainer information. We can cover this in next week's telecon, or before as appropriate. Will put link in dev telecon notes.

**Comment by ksbeattie** [comment]
@dangunter is this ready for review or is there more work to be done here?

**Comment by dangunter** [comment]
@ksbeattie it is ready for review. I will add you as a reviewer

**Comment by ksbeattie** [comment]
@dangunter there are now, unfortunately, quite a few conflicts here now...

**Comment by ksbeattie** [comment]
@dangunter this is looking increasingly like it won't make it into the Aug release...

**Comment by dangunter** [comment]
ok @lbianchi-lbl and @ksbeattie and @andrewlee94 if you can review this quick before more conflicts arise...

File: idaes_examples/notebooks/docs/diagnostics/diagnostics_toolbox_doc.ipynb [hunk 1]
```diff
@@ -74,331 +77,7 @@
           "solution"
         ]
       },
-      "outputs": [
```
[the remark this answers is not in the document]
Comment (dangunter) [reply idaes_examples/notebooks/docs/diagnostics/diagnostics_toolbox_doc.ipynb:77]: yeah sorry something changed or broke

[the remark this answers is not in the document]
Comment (dangunter) [reply idaes_examples/notebooks/docs/diagnostics/diagnostics_toolbox_doc.ipynb:77]: check again. output is backl!

**Review by lbianchi-lbl — APPROVED** [review]

**Pull request merged**